ToolBox/TiingoNewsConverter/TiingoNewsConverter.cs

```diff
@@ -0,0 +1,301 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using QuantConnect.Util;
+using QuantConnect.Data.Custom.Tiingo;
+using QuantConnect.Logging;
+using Newtonsoft.Json.Linq;
+
+namespace QuantConnect.ToolBox.TiingoNewsConverter
+{
+    public class TiingoNewsConverter
+    {
+        private const int TaskCountLimit = 200;
+        private readonly DirectoryInfo _sourceDirectory;
+        private readonly DirectoryInfo _rootDestinationDirectory;
+        private readonly DirectoryInfo _contentDirectory;
+        // date to process, if null will process all data
+        private readonly DateTime? _date;
+
+        /// <summary>
+        /// Creates an instance of the converter
+        /// </summary>
+        /// <param name="sourceDirectory">Directory to read raw data from</param>
+        /// <param name="destinationDirectory">Directory to write processed data to</param>
+        /// <param name="date">The date we want to process, if null will process all data</param>
+        public TiingoNewsConverter(DirectoryInfo sourceDirectory, DirectoryInfo destinationDirectory, DateTime? date = null)
```

Comment:
Interesting, we can process the full historical data if we don't declare a date.